src/include/sof/audio/component.h

```diff
@@ -249,6 +249,14 @@ struct comp_dev {
 	uint32_t output_rate;      /**< 0 means all output rates are fine */
 	struct pipeline *pipeline; /**< pipeline we belong to */
 
+	uint32_t min_sink_bytes;   /**< min sink buffer size measured in bytes
+				     *  required to run module
+				     */
+	uint32_t min_source_bytes; /**< amount of data measured in bytes
+				     *  available at source buffer required
+				     *  to run module
+				     */
+
```

Comment:
@bkokoszx what do you understand by "module" here. We are contemplating on adding a new component type named *module*.

A module could be for example implementing an algorithm (mp3, etc) which takes some input frames does some processing and outputs the processed output frames. 

Cc: @paulstelian97

Reply:
@bkokoszx i see now! Please also update the commit message :).

Reply:
@dbaluta
I've update code's comments and commit message. 

I have also meant components which implements some processing algorithms (audio codecs etc.) - they consume min_source_bytes at input and produce min_sink_bytes at output. Do we need  separate component type? :) We have for example ```struct sof_ipc_comp_process```.

Reply:
@bkokoszx seems that sof_ipc_comp_process is what we need. Any idea what are the values possible for 

```      uint32_t type;» /**< sof_ipc_effect_type */ ```

I couldn't find any definition for sof_ipc_effect type.

Also, do you really want this to be merged right now? I'm usually against adding structures for which there isn't yet any code to use them. While actually working on code using them you might figure out that you must do further changes.

Reply:
Component, which utilizes ```struct sof_ipc_comp_process``` is e.g. kpb (```SOF_COMP_KPB``` type), eq_fir (```SOF_COMP_EQ_FIR``` type), selector (```SOF_COMP_SELECTOR``` type). For new component, we should add new, specific type id. 
We can wait with merge until there will be some more comments and opinions under this PR :)

Reply:
@bkokoszx OK, thanks. Comments for sof_ipc_comp_process are not aligned between SOF and Kernel. Someone needs to fix the comment for SOF. It is a good first task.

Reply:
@dbaluta 
What do you mean by "Comments describing members of struct sof_ipc_comp_process are not aligned between SOF and Kernel". I've just looked at sof_ipc_comp_process struct (ans comments) in FW and Linux and they seem to be the same. :)

Reply:
@bkokoszx I might be having an older code? Can you please check this: https://github.com/thesofproject/sof/pull/2076